client/state/plugins/wporg/selectors.js

```diff
@@ -1,27 +1,39 @@
-const getPlugin = function( state, pluginSlug ) {
-	if ( ! state || ! state[ pluginSlug ] ) {
+export function getPlugin( state, pluginSlug ) {
+	let plugins;
+	// Some components just pass in the list of plugins, not the entire state.
+	if ( ! state.hasOwnProperty( 'plugins' ) ) {
+		plugins = state;
+	} else {
+		plugins = state.plugins.wporg.items;
+	}
+	if ( typeof plugins[ pluginSlug ] === 'undefined' ) {
 		return null;
 	}
-	return Object.assign( {}, state[ pluginSlug ] );
-};
+	return Object.assign( {}, plugins[ pluginSlug ] );
```

Comment:
Not related with this PR, but is there are reason behind returning a copy of the state leaf in the selectors? I haven't seen this being done a lot, and there was a discussion (p1475751304001205-slack-calypso-framework) about the fact that this is bad for performance (selectors should be fast), and we should have a mindset for avoiding state mutations anyway.